_posts/2024/2024-02-06-our-process.md

```diff
@@ -0,0 +1,453 @@
+    -   Which of these needs is most pressing to solve?
+-   Frequency: ongoing
+-   Responsible: Product Manager
+-   Contributors: technical leads and selected users
+-   Products:
+    -   Prioritized lists of problems
+    -   Prioritized lists of features or fixes that would solve them
+
+### Solution Discovery
+
+This activity is where product management and engineering design overlap.
+Its output is the starting point for the design docs mentioned earlier:
+in particular,
+this is where needs, constraints, and big-picture alternatives are formulated.
+
+-   Summary: Exploration and validation of potential solutions to prioritized problems
+-   Key Questions:
+    -   Should we build it or buy it?
+    -   How could we solve this problem for our users?
+    -   Which of these solutions seems most promising and are worth further exploration?
+    -   What risks and assumptions are inherent to these solutions?
+    -   What is the fastest, cheapest way to validate these solutions?
+-   Frequency: ongoing
+-   Responsible: Product Manager and technical leads
+-   Contributors: users and engineering team members
+-   Products:
+    -   Designs for new features or applications
+
+### Engineering Needs
+
+Software engineers are people too,
+and have their own specialized needs.
+Our product manager occasionally treats us as users
+and uses discovery techniques to (help us) figure out
+what would make *our* lives better.
+
+-   Summary: Ongoing assessment of engineering's needs
+-   Key Questions:
+    -   What parts of our tools, systems, architecture are slowing us down?
+    -   What new technical advancements should we be aware of?
+    -   What do we *need* vs. what do we *want*?
+-   Frequency: ongoing
+-   Responsible: technical leads
+-   Contributors: entire engineering team
+-   Products:
+    -   Designs for new features or applications
+
+### Biweekly Sprint Goals/Backlog Grooming
+
+OK,
+we've agreed on what we're going to build—now we actually have to build it.
+Every two weeks,
+the Scrum master and the project's technical lead
+create issues for new work
+and (if necessary) relabel old issues
+to define what we're going to try to accomplish in this sprint.
+Most of the new issues come from the design docs,
+or are elaborations or breakdowns of things discussed in those docs;
+depending on the size of the work,
+we may create several sub-issues under a single high-level issue
+so that progress reports aren't "I'm working on it" several days in a row.
+
+-   Summary: Biweekly prioritization to determine team focus for the sprint
+-   Key Questions:
+    -   What is the next slice of work we want to deliver?
+    -   What small improvements should we prioritize?
+    -   What key bugs should we prioritize?
+    -   What tech debt needs to be addressed?
+-   Frequency: biweekly
+-   Responsible: Scrum master and technical leads
+-   Contributors: engineering team members
+-   Products:
+    -   Labeled issues
```

Comment:
Slight tweak: Prioritized issues and goals for delivery

Reply:
reworded

Reply:
and expanded